x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -182,8 +185,15 @@ public SnapshotInfo getSnapshotInfo(SnapshotId snapshotId) {
         ArrayList<String> indices = new ArrayList<>(indicesMap.size());
         indicesMap.keysIt().forEachRemaining(indices::add);
 
-        return new SnapshotInfo(snapshotId, indices, new ArrayList<>(metadata.dataStreams().keySet()), Collections.emptyList(),
-            response.getState().getNodes().getMaxNodeVersion(), SnapshotState.SUCCESS
+        context.onResponse(
```

Comment:
Oh oops I just saw this. Think we need to take steps to put this on the `SNAPSHOT_META` thread now.

Reply:
timing https://github.com/elastic/elasticsearch/pull/73570#issuecomment-860731736 :D on it shortly

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/73570/commits/94f44ed1050cdfa38cc99e094dd4a0012b03b8c5 now